### DISCUSSION TITLE
Enhancement for Instance Service's Reference implementation

### DISCUSSION THREAD

**Discussion by johannesheucher-gip** [Xyna-Factory/xyna-factory#489] [Ideas]
## Now
Implementing a reference Workflow requires the user to model the complete signature (including throws) inside the "Abstract" section, before he can see the desired Workflow options inside the dropdown.

### Problems
* Modelling the signature inside the "Abstract" section is not intuitive (and implicitly sets the abstract flag on the data type)
* Having to model the complete signature needs much time and is not very adaptive to signature changes

## Proposal
The dropdown offers all Workflows that have the owning data type as its first input (regardless of output and throws signature).
Having selected a Workflow also establishes this Service's signature (must be displayed - readonly).
We also need an "Open" button besides the dropdown to open the reference Workflow.

### Sub Types
For an overridden Service, it is not possible to change the signature except for the following cases:
1. The input variables may be more general
2. The output variables may be more specific
3. The exceptions may be in a subset

For example, this has an effect for the provided reference Workflows.

**Comment by TorbenSiegismund-GIP** [comment]
A button to create a new Workflow with the correct signature next to the "Open" button could be a nice addition.
